### ISSUE TITLE
Servo Limits

### ISSUE THREAD

**Issue by BzowK** [3KUdelta/ESP32_ServoClock_MST#3]
Hey - 

I'm sorry to bother again, but have one final issue I cannot figure out for the life of me.  I've got the panels fine tuned as best possible, however; when some of them are off, I can hear a few of the servos seizing as their panels have hit the frame.  Adjusting them via finetuning so they do not do this has them go out of alignment and I'm certain they are seated correctly so am trying to set the global "off" position.

I tried adjusting the "210" & "450" values of "const int servopulse [2] {210, 450};" yet doing so affects the reverse servos in the opposite direction as desired.  Is this a code change that can easily be achieved by only changing a couple of values?

Thanks again!

**Comment by 3KUdelta** [comment]
Hi, 
I am not sure that the problem lays in the software. I assume a hardware issue. Please check these things: 1.) do you provide enough power? You need at least 3 amps. 2.) do you power the PCA9685 over the power plug on top and not over the solder points? 3.) are you using en ESP32? I had trouble using an ESP8266. 4.) Are the cables properly soldered (connected)? 5.) avoid taking 5 v from the ESP - simply follow exactly the drawing.
Hope this helps.

**Comment by BzowK** [comment]
Thanks for the reply...

Below are answers, but I don't believe it's pwoer-related.  It's not that the servos don't have enough power - they do.  It's that when some rotate into the off position (especially the bottom for each #), their attached "segment" hits the frame preventing it from reaching it's desired position.  When this occurs, the servo keeps trying to move it there, but can't therefore causing the buzz and making the servo get hot over time.

Hope that explains it.  Below is the answer for each question just in case.  Thanks!

1. Yes - 5V 3A source (via UBEC)
2. Correct - the top of each PCA9685 is connected directly to the 5V 3A
3. I'm using a NodeMCU ESP-12E which is fed via 5V 3A + capacitor as suggested in your diagram
4. Yes - everything is soldered
5. I did the diagram exactly power-wise

**Comment by 3KUdelta** [comment]
Good to read that it is not power related. I woiuld help a great deal if you could send a link to a video to better understand what is happening. And please add the part of the code with the settings.

**Comment by 3KUdelta** [comment]
https://photos.app.goo.gl/M7U6YdA9xsjYVASb8
This is how it should look like.

**Comment by BzowK** [comment]
Hello - 

Thanks again for your reply and willingness to assist.  I took a video a few minutes ago which is linked below.  Once the panels turn off, you should be able to hear the buzzing from the servos.  At the end of the video I go up close to the bottom (G) one of the first digit where you can see the panel hitting the frame which blocks the rotation and causes the buzzing.  This seems to affect most if not all panels on the top (A) & bottom (G).  You may notice these panels misaligned when all are on - this is because I've been trying to fine-tune them so they won't hit the frame without success.
[https://photos.app.goo.gl/JWaNE2e1TMyQ16pVA](https://photos.app.goo.gl/JWaNE2e1TMyQ16pVA)

Below is my current sketch.  The only changes should be from me testing things, adding in code to convert to a 12-hour clock, & code to display the hour to be "12" when reported as "00".
[https://kellermannnetwork.com/owncloud/s/lAuKBkuCU0jOKZB](https://kellermannnetwork.com/owncloud/s/lAuKBkuCU0jOKZB)

Hope that helps - any suggestions?  Thanks again!

**Comment by 3KUdelta** [comment]
One thing I can spot immediately. The middle servos are in the wrong "park" positions. If they are not showing the bar they are supposed to go downwards. Yours are going upwards.

**Comment by BzowK** [comment]
Hmm - That's odd. Just looked and none of them are surging / hitting the frame unfortunately...

**Comment by 3KUdelta** [comment]
OK, got some time to go through your things. 
Just tweak the following line and you should be ok:
`const int servopulse [2] {210, 450};      // pulse of servos at low positions and high position`
The values 210 and 450 give the two different up and down states. So you might change 450 to 430 or less to get rid of your problem.

**Comment by BzowK** [comment]
Good Day -

Thanks again.  Will try out in a bit and let you know results.  Still wondering why middle servos are flipping opposite from yours, too.  If nogo, may start fresh from you uploaded sketch again and see results.

Thanks

**Comment by 3KUdelta** [comment]
Hi, I don't think that matters a lot. As long as the individual servos are not blocking each other. 
You can change this here:
`const byte servoreverse [4] [7] = {      // one identifies wservos that work reverse, zero is normal direction
  {1, 1, 0, 0, 0, 1, 1},
  {1, 1, 0, 0, 0, 1, 1},
  {1, 1, 0, 0, 0, 1, 1},
  {1, 1, 0, 0, 0, 1, 1}
};`
First things first - change the high/low positions and you will be fine...

**Comment by BzowK** [comment]
Good Day - 

Think that did it!  I changed it to "400".  Could have sworn I tried that previously, but have probably tried well over 100 sketch modifications so may had missed something in the past or more than likely not adjusted it enough.  Thanks!

To note, there are still two things I had to do initially from your code to get things working in case you wish to update them...

- I compared my sketch with the one you have posted and remembered that initially it wouldn't even compile without adding a lines.  Prior to the "void get_NTP_time() {" line, I had to add "void(* resetFunc) (void) = 0; //declare reset function @ address 0
" else it wouldn't compile
- I also had to swap the SLA / SLC from how you had them wired in the diagram.

Just notes in case you wish to change.  Thanks again for your help!
